foreshadow/utils/common.py

```diff
@@ -96,15 +98,57 @@ def get_transformer(class_name, source_lib=None):
     return getattr(module, class_name)
 
 
-class ConfigureColumnSharerMixin:
-    """Mixin that configure column sharer."""
+class ConfigureCacheManagerMixin:
+    """Mixin that configure cache_manager."""
 
-    def configure_column_sharer(self, column_sharer):
+    def configure_cache_manager(self, cache_manager):
         """Configure the column sharer attribute if exists.
 
         Args:
-            column_sharer:  a column sharer instance
+            cache_manager:  a cache_manager instance
 
         """
-        if hasattr(self, "column_sharer"):
-            self.column_sharer = column_sharer
+        if hasattr(self, "cache_manager"):
+            self.cache_manager = cache_manager
+
+
+class UserOverrideMixin:
```

Comment:
Where are we using this?

Reply:
The UserOverrideMixin is implemented (or inherited) by SmartTransformers to determine whether they should force reresolve due to user override. This allows the DataCleaners to define their own rule instead of mixing up with the intent override logic.